### ISSUE TITLE
Need Dev Enviro Checklist

### ISSUE THREAD

**Issue by parispittman** [kubernetes/community#1824]
Need to add dev environment checklist for new contributors - how to set up on multiple platforms and address main issues / concerns. Add to new contributor guide. 

/sig contributor-experience
/area contributor-guide

**Comment by nikhita** [comment]
+1. More explicit details in https://github.com/kubernetes/community/blob/master/contributors/devel/running-locally.md would be really, really helpful.

I've usually gotten the feedback that people don't want to go to each dependencies' page and checkout the install process. They usually expect "do this, this and this and boom - you are done!". Hopefully accomplishing this is feasible enough (haven't looked too deep into it).

**Comment by nikhita** [comment]
ref https://github.com/kubernetes/community/issues/505#issuecomment-324139042

**Comment by carlpett** [comment]
Having just gone though this process myself - yes, something clearer would be very helpful! I didn't really want to risk my entire laptop by running my own poorly understood code as root (wanted to make a change in storage-related bits), so I ended up using Vagrant to have a disposable cluster VM. 
This approach is sort of implied in the docs, they mention minikube but not really how to get your build output into minikube.
My Vagrant approach is still somewhat rough around the edges, but I appreciated not having to install a lot of dependencies on my OS, as well as not accidentally using my local drive for a pod disk...
